src/cmds/scripts/pbs_habitat.in

```diff
@@ -240,19 +240,19 @@ upgrade_pbs_database() {
 	fi
 	
 	sys_pgsql_ver=$(echo `${PGSQL_DIR}/bin/psql -V` | awk '{print $NF}' | cut -d '.' -f 1,2)
-	ver=`cat ${data_dir}/PG_VERSION`
+	old_pgsql_ver=`cat ${data_dir}/PG_VERSION`
 
-	[ ${ver%.*} -eq ${sys_pgsql_ver%.*} ] && [ ${ver#*.} \> ${sys_pgsql_ver#*.} ] || [ ${ver%.*} -gt ${sys_pgsql_ver%.*} ];
+	[ ${sys_pgsql_ver%.*} -eq ${old_pgsql_ver%.*} ] && [ ${sys_pgsql_ver#*.} \> ${old_pgsql_ver#*.} ] || [ ${sys_pgsql_ver%.*} -gt ${old_pgsql_ver%.*} ];
 	result=$?
 	if [ ${result} -eq 0 ]; then
 		#Start upgrade process of datastore
 		upgrade_db
 		return $?
-	elif [ "${ver}" = "${sys_pgsql_ver}" ]; then
+	elif [ "${old_pgsql_ver}" = "${sys_pgsql_ver}" ]; then
```

Comment:
That part helps with determining whether the major version is the same, and only then checks for minor version. So without the first check, it would exit saying "Unsupported" when we have same major release number with smaller minor release number, but would try to upgrade for smaller major release number and greater minor release number.

For example, if we remove the first equality check, when sys_pgsql_ver=8.9 and old_pgsql_ver=9.2, it will call upgrade_db. But when sys_pgsql_ver=9.1 and old_pgsql_ver=9.2, it will exit saying "Upgrade from version 9.2 unsupported" (which is what should happen).